src/main/java/io/confluent/kafkarest/resources/v2/ConsumersResource.java

```diff
@@ -0,0 +1,310 @@
+  @PerformanceMetric("consumer.create+v2")
+  public CreateConsumerInstanceResponse createGroup(
+      @javax.ws.rs.core.Context UriInfo uriInfo, final @PathParam("group") String group,
+      @Valid ConsumerInstanceConfig config) {
+    if (config == null) {
+      config = new ConsumerInstanceConfig();
+    }
+    String instanceId = ctx.getKafkaConsumerManager().createConsumer(group, config);
+    String instanceBaseUri = UriUtils.absoluteUriBuilder(ctx.getConfig(), uriInfo)
+        .path("instances").path(instanceId).build().toString();
+    return new CreateConsumerInstanceResponse(instanceId, instanceBaseUri);
+  }
+
+  @DELETE
+  @Path("/{group}/instances/{instance}")
+  @PerformanceMetric("consumer.delete+v2")
+  public void deleteGroup(final @PathParam("group") String group,
+                          final @PathParam("instance") String instance) {
+    ctx.getKafkaConsumerManager().deleteConsumer(group, instance);
+  }
+
+  @POST
+  @Path("/{group}/instances/{instance}/subscription")
+  @PerformanceMetric("consumer.subscribe+v2")
+  public void subscribe( @javax.ws.rs.core.Context UriInfo uriInfo, 
+			 final @PathParam("group") String group,
+			 final @PathParam("instance") String instance,
+			 @Valid ConsumerSubscriptionRecord subscription
+			) {
+      try {
+	  ctx.getKafkaConsumerManager().subscribe(group, instance, subscription);
+      } catch(java.lang.IllegalStateException e) {
+	  throw Errors.illegalStateException(e);	  
+      } 
+   
+  }
+
+  @GET
+  @Path("/{group}/instances/{instance}/subscription")
+  @PerformanceMetric("consumer.subscription+v2")
+  public ConsumerSubscriptionResponse subscription( @javax.ws.rs.core.Context UriInfo uriInfo, 
+			 final @PathParam("group") String group,
+			 final @PathParam("instance") String instance
+			) {
+      return ctx.getKafkaConsumerManager().subscription(group, instance);      
+  }
+
+  @DELETE
+  @Path("/{group}/instances/{instance}/subscription")
+  @PerformanceMetric("consumer.unsubscribe+v2")
+  public void unsubscribe( @javax.ws.rs.core.Context UriInfo uriInfo, 
+			 final @PathParam("group") String group,
+			 final @PathParam("instance") String instance
+			) {
+      ctx.getKafkaConsumerManager().unsubscribe(group, instance);      
+  }
+    
+  @GET
+  @Path("/{group}/instances/{instance}/records")
+  @PerformanceMetric("consumer.records.read-binary+v2")
+  @Produces({Versions.KAFKA_V2_JSON_BINARY_WEIGHTED,
+             Versions.KAFKA_V2_JSON_WEIGHTED
+	      })
+  public void readRecordBinary(final @Suspended AsyncResponse asyncResponse,
+                              final @PathParam("group") String group,
+                              final @PathParam("instance") String instance,
+		              @QueryParam("timeout") @DefaultValue("-1") long timeout,
+                              @QueryParam("max_bytes") @DefaultValue("-1") long maxBytes) {
+      readRecords(asyncResponse, group, instance, timeout, maxBytes, BinaryKafkaConsumerState.class);
```

Comment:
If the request doesn't contain a timeout, default value **-1** is passed and no other default is set, which trickles down to consumer.poll() in KafkaConsumerState.getOrCreateTopicState(), as per the Consumer code in KafkaConsumer, it will throw a **IllegalArgumentException("Timeout must not be negative")** . More sensible default must be set

Similar thing with max_bytes, if the request doesn't set any, then a **-1** is set, in V1 ConsumersResource it does a **maxBytes = (maxBytes <= 0) ? Long.MAX_VALUE : maxBytes;**, similar can be done here